mne/viz/ica.py

```diff
@@ -130,6 +130,255 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
+        and epochs image use their default "RdBu_r" colormap. Defaults to None.
+    plot_std: bool | float
+        Whether to plot standard deviation in ERP/ERF and spectrum plots.
+        Defaults to True, which plots one standard deviation above/below.
+        If set to float allows to control how many standard deviations are
+        plotted. For example 2.5 will plot 2.5 standard deviation above/below.
+    topo_args : dict | None
+        Dictionary of arguments to plot_topomap. If None - doesn't pass any
+        additional arguments. Defaults to None.
+    image_args : dict | None
+        Dictionary of arguments to plot_epochs_image. If None - doesn't pass
+        any additional arguments. Defaults to None.
+    psd_args : dict | None
+        Dictionary of arguments to psd_multitaper. If None - doesn't pass
+        any additional arguments. Defaults to None.
+    figsize : array-like of size (2,) | None
+        Allows to control size of the figure. If None the figure size
+        defauls to [7., 6.].
+    show : bool
+        Show figure if True.
+
+    Returns
+    -------
+    fig : instance of pyplot.Figure
+        The figure.
+    """
+    from .epochs import plot_epochs_image
+    from ..io.base import _BaseRaw
+    from ..epochs import _BaseEpochs
+    from ..preprocessing import ICA
+    from ..time_frequency.psd import psd_multitaper
+
+    # current component/channel (with future interactivity in mind)
+    idx = 0
+
+    if not isinstance(inst, (_BaseRaw, _BaseEpochs)):
+        raise ValueError('inst should be an instance of Raw or Epochs,'
+                         ' got %s instead.' % type(inst))
+    if not isinstance(ica, ICA):
+        raise ValueError('ica has to be an instance of ICA, '
+                         'got %s instead' % type(ica))
+    else:
+        from .topomap import _plot_ica_topomap
+    if isinstance(plot_std, bool):
+        num_std = 1. if plot_std else 0.
+    elif isinstance(plot_std, (float, int)):
+        num_std = plot_std
+        plot_std = True
+    else:
+        raise ValueError('plot_std has to be a bool, int or float, '
+                         'got %s instead' % type(plot_std))
+
+    # if no picks given - plot the first 5 components
+    picks = list(range(min(5, ica.n_components_))) if picks is None else picks
+    picks = [picks] if isinstance(picks, int) else picks
+    if axes is None:
+        fig, axes = _create_properties_layout(figsize=figsize)
+    else:
+        if len(picks) > 1:
+            raise ValueError('Only a single pick can be drawn '
+                             'to a set of axes.')
+        from .utils import _validate_if_list_of_axes
+        _validate_if_list_of_axes(axes, obligatory_len=5)
+        fig = axes[0].get_figure()
+    psd_args = dict() if psd_args is None else psd_args
+    topo_args = dict() if topo_args is None else topo_args
+    image_args = dict() if image_args is None else image_args
+    for d in (psd_args, topo_args, image_args):
```

Comment:
Actually not - `d = dict() if d is None else d` would bound the value to the temporary `d` variable, so it has to stay this way (unless we want an `eval` there)